java/pgv-artifacts/pom.xml

```diff
@@ -0,0 +1,217 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>pgv-java</artifactId>
+        <groupId>com.lyft.protoc-gen-validate</groupId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>protoc-gen-validate</artifactId>
+    <packaging>pom</packaging>
+
+    <name>PGV-Java Protoc Plugins</name>
+
+    <profiles>
+        <profile>
+            <id>public-release</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>exec-maven-plugin</artifactId>
+                        <version>1.6.0</version>
+                        <executions>
+                            <execution>
+                                <id>osx-x86_64</id>
+                                <goals>
+                                    <goal>exec</goal>
+                                </goals>
+                                <phase>compile</phase>
+                                <configuration>
+                                    <executable>go</executable>
+                                    <arguments>
+                                        <argument>build</argument>
+                                        <argument>-o</argument>
+                                        <argument>${project.build.directory}/protoc-gen-validate-${project.version}-osx-x86_64.exe</argument>
```

Comment:
isn't it a bit atypical to suffix *nix executables with .exe?

Reply:
Normally, yes.

However, Maven is funny. All artifacts must have a type, which corresponds to a file extension. The default type is `.jar`. This means all artifacts must have a file extension, even if they are executables.

I talked with the gRPC team about how they distribute the gRPC-java protoc plugin. They use the same pattern.